### ISSUE TITLE
Every time the browser is started, uBlock Origin connects to an external server without asking the user

### ISSUE THREAD

**Issue by dsh4dtw** [uBlockOrigin/uBlock-issues#1641]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported
- [x] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
    - If the issue is still present after wholly disabling uBO in the browser, then the issue is unrelated to uBO
- I tried to reproduce the issue when...
    - [x] uBO is the only extension
    - [x] uBO with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

uBlock Origin automatically connects to external servers each time the browser is started without asking the user.

### A specific URL where the issue occurs
These are the external servers that uBlock Origin connects to each time the browser is started.

- d2nxq2uap88usk.cloudfront.net
- content-signature-2.cdn.mozilla.net

### Steps to Reproduce

1. Install uBlock Origin locally for the first time and add all the lists you want locally. Disable Auto-Update filter list under Filter lists.
2. Exit the browser
3. Connect to an internet connection
4. Start the browser
5. uBlock Origin will automatically connect to external server without asking the user
6. There is easy way for user to disable this behavior

### Expected behavior:

If users have everything installed locally and all options disabled in the UI that are not local options, then uBlock Origin must not automatically connect to external servers without asking the user.

### Actual behavior:

uBlock Origin automatically connects to external servers without asking the user. 

### Your environment

* uBlock Origin version: 1.36.0
* Browser Name and version: Customized Firefox 89.0.1, which is designed to behave as expected. That is, everything that the user would generally see in the UI is designed so that you know it's a local option or an option that requires an internet connection. There are no hidden background servers or invisible options that are not displayed in the UI.
* To summarize:
By default, my customized Firefox version does not connect to any external server.

* Operating System and version: Linux Kernel 5.12.12 and customized desktop environment

**Comment by uBlock-user** [comment]
> content-signature-2.cdn.mozilla.net

Firefox connects to this server to authenticate extensions signature with their server, not uBO.

> d2nxq2uap88usk.cloudfront.net

Unknown

**Comment by gorhill** [comment]
> add all the lists you want locally

And what are those "all the lists"? Provide details otherwise there is nothing which can be investigated.

If you subscribe to filter lists which are not part of the package, uBO will have to fetch them from the server on which they are hosted if you want them enforced.

> Connect to an internet connection

Was there an internet connection when you added "all the lists"?

**Comment by gorhill** [comment]
> `content-signature-2.cdn.mozilla.net`

That's not uBO, it's Firefox probably verifying the signature.

**Comment by dsh4dtw** [comment]
You are both right, I have tried several times and since I disabled uBlock Origin, this server still exists. So it is a Firefox problem after all.

I then found a possible bug.

uBlock Origin Settings -> Reset to default settings...

Whenever I click on this function, it always appears that it automatically connects to this external server -> **easylist.to**. Can you reproduce it on your site?

**Comment by gorhill** [comment]
host content-signature-2.cdn.mozilla.net
    content-signature-2.cdn.mozilla.net is an alias for d2nxq2uap88usk.cloudfront.net.

**Comment by gorhill** [comment]
> Whenever I click on this function

This resets to default settings, and hence auto-updated is reset to being enabled.

**Comment by ChiefMikeK** [comment]
> 6. There is easy way for user to disable this behavior  
---
If it's mozila telemetry then try **some** of this `user.js`  
`https://github.com/pyllyukko/user.js#firefox-anti-features--components`  

- Please ask for any futher assist on reddit r/firefox

**Comment by gorhill** [comment]
@ dsh4dtw 

> - [x] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser 

Then:

> You are both right, I have tried several times and since I disabled uBlock Origin, this server still exists.
